lib/tier-node.js

```diff
@@ -84,23 +84,54 @@ function handler (handle, params, req, res, body, query) {
   }
 
   const request = new Request(params, req, body, query)
-  handle.handler(request, reply)
+  handle.handler(request, new Reply(res, handle))
+}
 
-  function reply (err, statusCode, data) {
-    if (err) {
-      res.statusCode = 500
-      res.end()
-      return
+function Reply (res, handle) {
+  this.res = res
+  this.handle = handle
+}
+
+Reply.prototype.send = function (payload) {
+  if (payload instanceof Error) {
+    if (!this.res.statusCode) {
+      this.res.statusCode = 500
     }
+    this.res.end()
+    return
+  }
 
-    if (!data) {
-      data = statusCode
-      statusCode = 200
+  if (!payload) {
+    if (!this.res.statusCode) {
+      this.res.statusCode = 204
     }
```

Comment:
forget this. Add a very explicit return with a comment 'do not chain after send'.